core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -227,6 +228,17 @@ private[spark] class Executor(
               logError(errMsg)
             }
           }
+
+          if (releasedLocks.nonEmpty) {
+            val errMsg =
+              s"${releasedLocks.size} block locks were not released by TID = $taskId:\n" +
```

Comment:
...which would tend to argue that this should be named unreleasedLocks, not releasedLocks.